crates/file/src/blob.rs

```diff
@@ -0,0 +1,60 @@
+use wasm_bindgen::UnwrapThrowExt;
+
+pub trait BlobLike {
+    fn size(&self) -> usize {
+        self.as_raw().size() as usize
+    }
+
+    #[cfg(feature = "mime")]
+    fn mime_type(&self) -> Option<mime::Mime> {
+        Some(self.as_raw().type_().parse().ok()?)
```

Comment:
I think we should return the `Result` that `parse()` returns, rather than converting it to an `Option`